src/stravavis/plot_landscape.py

```diff
@@ -28,8 +28,13 @@ def plot_landscape(df, output_file = 'landscape.png'):
     for activity in track(activities, "Plotting activities"):
         x = df[df['name'] == activity]['dist_norm']
         y = df[df['name'] == activity]['ele']
-        plt.fill_between(x, y, color='black', alpha=0.03, linewidth=0)
-        plt.plot(x, y, color='black', alpha=0.125, linewidth=0.25)
+        if y.max() - y.min() > 4000 or y.min() < -1000 or y.max() > 9000:
+            continue
+        try:
+            plt.fill_between(x, y, color='black', alpha=0.03, linewidth=0)
+            plt.plot(x, y, color='black', alpha=0.125, linewidth=0.25)
+        except Exception as e:
```

Comment:
It's best to avoid catching `Exception`, it's overly broad and may hide other valid errors. 

What's the actual exception you're getting here? Can we catch that specific one instead?

Reply:
I guess it would be fine to remove this exception here, as I have checked `y` above.